### DISCUSSION TITLE
RFC: Default device in volume control

### DISCUSSION THREAD

**Discussion by LBCrion** [LBCrion/sfwbar#292] [Ideas]
I'm looking for opinions on meaning and handling of default devices in the volume control widget. (See #290). Historically, sfwbar used terms 'default sink'/'default source' to denote the device sfwbat would control by default. I.e. if you scroll over the volume widget icon, the volume will change on the devices designated as default.

Sfwbar default device, would normally track the pulse default device (i.e. the sink or source that applications would use by default). However, if the user explicitly selects one device to be the default, controls will use this device, even if pulse default sink/source changes. This can be useful, if you have a specific application playing sound on a non-default sink and you want to control volume to this application easily. 

In #290 we introduced ability to set pulse default devices from sfwbar. I.e. the device that applications will use by default. This brings up a question of how to make UI sensibly represent the two concepts of default. Currently, I have kept the widget as is, adding the new functionality through a right click menu attached to the sink name in the widget.  (i.e. the big, prominent checkboxes continue to select default sfwbar control device, while the right click menu selects the default pulse device).

I suspect most people care much more about the default pulse sink/source than they do about the default sfwbar control device. If this is the case, it would make sense to reverse the interface elements and make the big checkboxes select the default pulse device and right click menu select the default sfwbar control device.

Appreciate any thoughts / opinions on this.

**Comment by Flrian** [comment]
There's not much I can add, I only have one audio device and don't really have a use for the feature currently. I just wanted to say I always thought the default device would be the pulse default device, so the naming feels indeed a bit confusing.
